### ISSUE TITLE
[Bug]: macOS 15.2 首次安装 ComfyUI-Desktop 失败的问题

### ISSUE THREAD

**Issue by dong4j** [Comfy-Org/desktop#483]
### App Version

This time I tried installing ComfyUI-Desktop for the second time, version 0.3.33(241212). The automated install didn't work. After checking the logs and manually running the missing commands, the installation was successful. I want to give friends who run into the same issue a workaround. I'm not sure if it's a bug, but it didn't deploy correctly on my machine. So next, I'll try to reproduce the problem in the hope that the official team can fix it soon.

### Expected Behavior

After startup, the automated deployment kicks in and successfully opens the UI page.

### Actual Behavior

Cannot automate installation and deployment.

### Steps to Reproduce

[Here's an issue I had with the first installation](https://github.com/Comfy-Org/desktop/issues/398), it seems that version 0.3.33 has fixed it, but I still run into problems under zsh, so this time I'll just use the initial .zshrc file:

```
export ZSH="$HOME/.oh-my-zsh"
ZSH_THEME="robbyrussell"
plugins=(git)
source $ZSH/oh-my-zsh.sh
```

Then try to start for the first time:

![CleanShot_2024121494PlSgGY](https://github.com/user-attachments/assets/158af3be-6d51-4cfb-b4c6-f2bad78e4cc2)

日志如下:

```
[2024-12-14 22:01:21.481] [info]  Running command: /Users/dong4j/comfy/ComfyUI/.venv/bin/python -m ensurepip --upgrade in /Users/dong4j/comfy/ComfyUI
[2024-12-14 22:01:23.353] [info]  Successfully created virtual environment at /Users/dong4j/comfy/ComfyUI/.venv
[2024-12-14 22:01:23.354] [info]  Installing PyTorch Nightly for macOS.
[2024-12-14 22:01:23.354] [info]  Running uv command: /Applications/ComfyUI.app/Contents/Resources/uv/macos/uv pip install -U --prerelease allow torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cpu
...
[2024-12-14 22:01:23.367] [info]  ]2;clear; /Applications/ComfyUI.app/Contents/Resources/uv/macos/uv "pip"  "-U"  ]1;clear;
[2024-12-14 22:01:23.369] [info]  
[2024-12-14 22:01:23.375] [info]  error: unexpected argument '--end-1734184883354:0' found

  tip: to pass '--end-1734184883354:0' as a value, use '-- --end-1734184883354:0'

Usage: uv pip install --upgrade --prerelease <PRERELEASE> <PACKAGE|--requirements <REQUIREMENTS>|--editable <EDITABLE>> <--index <INDEX>|--default-index <DEFAULT_INDEX>|--index-url <INDEX_URL>|--extra-index-url <EXTRA_INDEX_URL>|--find-links <FIND_LINKS>|--no-index>

For more information, try '--help'.
```

The key point is:

```
/Applications/ComfyUI.app/Contents/Resources/uv/macos/uv pip install -U --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cpu
```

It should be that there was an attempt to install [PyTorch](https://developer.apple.com/metal/pytorch/), but for some unknown reason it failed, so I directly used the python installed in the virtual environment (the directory set when starting ComfyUI-Desktop: Users/dong4j/comfy):

```bash
/Users/dong4j/comfy/ComfyUI/.venv/bin/pip3 install -U --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cpu
```

List of packages after installation:

```
➜  ComfyUI /Users/dong4j/comfy/ComfyUI/.venv/bin/pip3 list                                                                                                        
Package           Version
----------------- ------------------
filelock          3.16.1
fsspec            2024.10.0
Jinja2            3.1.4
MarkupSafe        3.0.2
mpmath            1.3.0
networkx          3.4.2
numpy             2.2.0
pillow            11.0.0
pip               24.0
setuptools        75.6.0
sympy             1.13.1
torch             2.6.0.dev20241214
torchaudio        2.6.0.dev20241214
torchvision       0.22.0.dev20241214
typing_extensions 4.12.2
```

Then restart ComfyUI-Desktop again:

![CleanShot_202412142KnSv16s](https://github.com/user-attachments/assets/f5f96164-718e-4369-a08c-b074ddf4c024)

It should be normal to proceed with the subsequent installation steps, but there are still missing packages. The list of packages at this time is as follows:

```
➜  ComfyUI /Users/dong4j/comfy/ComfyUI/.venv/bin/pip3 list
Package            Version
------------------ ------------------
certifi            2024.12.14
cffi               1.17.1
charset-normalizer 3.4.0
click              8.1.7
cryptography       44.0.0
Deprecated         1.2.15
filelock           3.16.1
fsspec             2024.10.0
gitdb              4.0.11
GitPython          3.1.43
huggingface-hub    0.26.5
idna               3.10
Jinja2             3.1.4
markdown-it-py     3.0.0
MarkupSafe         3.0.2
matrix-client      0.4.0
mdurl              0.1.2
mpmath             1.3.0
networkx           3.4.2
numpy              1.26.4
packaging          24.2
pillow             11.0.0
pip                24.0
pycparser          2.22
PyGithub           2.5.0
Pygments           2.18.0
PyJWT              2.10.1
PyNaCl             1.5.0
PyYAML             6.0.2
regex              2024.11.6
requests           2.32.3
rich               13.9.4
safetensors        0.4.5
setuptools         75.6.0
shellingham        1.5.4
smmap              5.0.1
sympy              1.13.1
tokenizers         0.21.0
torch              2.6.0.dev20241214
torchaudio         2.6.0.dev20241214
torchvision        0.22.0.dev20241214
tqdm               4.67.1
transformers       4.47.0
typer              0.15.1
typing_extensions  4.12.2
urllib3            1.26.20
wrapt              1.17.0
```

I tried to install dependencies directly using ComfyUI's `requirements.txt` this time (I had already tried installing dependencies manually step by step and successfully launched it before, following the prompts to install: psutil, einops, ,scipy, torchsde, aiohttp, spandrel, kornia [using `~/comfy/ComfyUI/.venv/bin/pip3 install xxx`]):

```bash
~/comfy/ComfyUI/.venv/bin/pip3 install -r /Applications/ComfyUI.app/Contents/Resources/ComfyUI/requirements.txt 
```

Next is the third launch:

![CleanShot_202412141qvmdwmj](https://github.com/user-attachments/assets/1ea4e799-daaf-4b46-bfab-a45132132b69)

![CleanShot_20241214wTL3f6bG](https://github.com/user-attachments/assets/151e374b-068e-4f3e-bee2-ff2698d97fd0)

Fortunately, the startup was successful. 
To summarize the issues: 

1. The installation of PyTorch failed, causing subsequent steps to be unexecutable; 
2. Other dependencies could not be automatically installed.

### Debug Logs

```powershell
无
```

### Browser Logs

无

### Setting JSON

无

**Comment by lips85** [comment]
I'm glad to hear it was helpful! If you have any further questions or need assistance, feel free to reach out anytime. 😊

**Comment by lips85** [comment]
I experienced exactly the same issue as you did!

Thank you!

**Comment by robinjhuang** [comment]
Thanks for reporting this. We currently add a string `--end-...` to the terminal output to know when the uv install ends. This was a hack to let us stream interactive shell output to our frontend. 

For now what if you run the command manually? 

```
cd <your chosen directory
/Applications/ComfyUI.app/Contents/Resources/uv/macos/uv pip install -U --prerelease allow torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cpu
```

Or just install pytorch in the .venv that was created

**Comment by lpffernando** [comment]
Same issue. How to solve? Any suggestions? Thanks.

**Comment by lpffernando** [comment]
> > Same issue. How to solve? Any suggestions? Thanks.
> 
> What happens when you run this command? Change YOUR_DIRECTORY to the location you chose at the start of the ComfyUI installer.
> 
> ```
> cd YOUR_DIRECTORY
> /Applications/ComfyUI.app/Contents/Resources/uv/macos/uv pip install -U --prerelease allow torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cpu
> ```

it works well when I use cmd ,and still not work when I use app, I think it's caused by the shell settings when start the app

**Comment by lpffernando** [comment]
> > it works well when I use cmd ,and still not work when I use app, I think it's caused by the shell settings when start the app
> 
> Yeah! That's exactly what we were testing.
> 
> It seems like the `echo` being run in your shell is rejecting the command because it does not like the `--` at the start of the first argument (despite the double-quotes, which is the weird part). Fix is in for this now and should be in next version.
> 
> For now, @dong4j mentioned a workaround about using the default `zsh` config - does temporarily using the default `.zshrc` work for you?

my .zshrc is simple, just conda initialize and proxy setting, the app may not use the proxy settings correctly...

**Comment by richtong** [comment]
Has this been fixed on 0.40. I'm having this hang with MacOS 15.2, but confused when you say CD into your DIRECTORY what you mean? What is the normal default? OH I see from the video, ComfyUI is really an installer, so you need a local installation location.

OK, really different from a typical Mac app, might want to consider changing the Mac name to "ComfyUIInstaller" :=)
